crates/turbo-tasks/src/read_ref.rs

```diff
@@ -130,17 +131,18 @@ impl<T, U: Serialize> Serialize for ReadRef<T, U> {
     }
 }
 
-impl<'de, T: Deserialize<'de>> Deserialize<'de> for ReadRef<T, T> {
+impl<'de, T, U: Deserialize<'de>> Deserialize<'de> for ReadRef<T, U> {
```

Comment:
```suggestion
impl<'de, T: Deserialize<'de>, U> Deserialize<'de> for ReadRef<T, U> {
```

You can deserialize the T instead of U to avoid the unsafe

Reply:
Verified that this works, because transparent Values have a trivial tuple deserializer to convert from U (the inner type) -> T (tuple type), and non-transparent Values are `T = U`.